ccdproc/image_collection.py

```diff
@@ -694,10 +700,12 @@ def _generator(self, return_type,
 
             file_name = path.basename(full_path)
 
+            extension_index = hdulist.index_of(self._extension)
+
             return_options = {
-                    'header': lambda: hdulist[0].header,
-                    'hdu': lambda: hdulist[0],
-                    'data': lambda: hdulist[0].data,
+                    'header': lambda: hdulist[extension_index].header,
+                    'hdu': lambda: hdulist[extension_index],
+                    'data': lambda: hdulist[extension_index].data,
                     'ccd': lambda: fits_ccddata_reader(full_path, **ccd_kwargs)
```

Comment:
I'd say if `ccd_kwargs` contains a `hdu`-keyword that should be used, otherwise one could add `hdu: extension_index` to the `ccd_kwargs`.